common/views/components/ArchiveTree/ArchiveTree.js

```diff
@@ -328,30 +311,42 @@ async function createArchiveTree({
   archiveAncestorArray: NodeWork[],
   toggles: Toggles,
 |}): Promise<UiTree> {
-  const allTreeNodes = [...archiveAncestorArray, parsePart(work)];
+  const allTreeNodes = [...archiveAncestorArray, parsePart(work)]; // An array of a work and all it's ancestors (ancestors first)
   const treeStructure = await allTreeNodes.reduce(
     async (acc, curr, i, ancestorArray) => {
-      const siblings =
-        i === allTreeNodes.length - 1
-          ? await getSiblingsWithDescendents({
-              id: curr.id,
-              toggles,
-              workId: work.id,
-              depth: 2,
-            })
-          : await getSiblingsWithDescendents({
-              id: curr.id,
-              toggles,
-              workId: work.id,
-            });
-
-      if (i === 0) {
-        return siblings || [];
+      const parentId = ancestorArray[i - 1] && ancestorArray[i - 1].id;
+      // We add each ancestor and it's siblings to the tree
+      if (!parentId) {
+        const siblings = await getSiblings({
+          id: curr.id,
+          toggles,
+          workId: work.id,
+        });
+        return siblings;
+      }
+      if (work.id === curr.id) {
+        // If it's the curr work we have all the information we need to create an array of it and its siblings
+        // This becomes the value of its parent node's children property
+        const siblings = createSiblingsArray({
+          work: work,
+          workId: work.id,
+        });
+        return updateChildren({
+          tree: await acc,
+          id: parentId,
+          value: siblings || [],
+        });
       } else {
-        const idOfObjectToUpdate = ancestorArray[i - 1].id;
+        const siblings = await getSiblings({
+          // For everything else we need more data and create before creating and array of curr and its siblings
```

Comment:
I don't follow this comment

Reply:
I don't blame you - sorry, note to self not to work late at night...